### ISSUE TITLE
Invalid code in MyStorage.php

### ISSUE THREAD

**Issue by phil-davis** [owncloud/docs#758]
## What Needs to be Documented?
The example code shows various warnings an errors in PHPstorm, like:

```
	public function getConnection() {
		if ($this->connection === null) {
			// do the connection to the storage lazily
			$this->connection = new \SomeVendor\FakeStorageLib\Connection($params);
		}
		return $this->connection;
	}
```
Undefined variable $params

```
	public function fopen($path, $mode) {
		switch ($mode) {
			case 'r':
			case 'rb':
				// this works if the library returns a PHP stream directly
				return $this->connection->getStream($path);
			case 'w':
			case 'w+':
			case 'wb':
			case 'wb+':
			case 'a':
			case 'ab':
			case 'r+':
			case 'a+':
			case 'x':
			case 'x+':
			case 'c':
			case 'c+':
				// most storages do not support on the fly stream upload for all modes,
				// so we use a temporary file first
				$ext = pathinfo($filename, PATHINFO_EXTENSION);
				$tmpFile = \OC::$server->getTempManager()->getTemporaryFile($ext);

				// this wrapper will call the callback whenever fclose() was called on the file,
				// after which we send the file to the library
				$result = CallbackWrapper::wrap(
					$source,
					null,
					null,
					function () use ($tmpFile, $path) {
						$this->connection->putFile($tmpFile, $path);
						unlink($tmpFile);
					}
				);
		}
		return false;
	}
```
Undefined variable $filename

```
	public function fopen($path, $mode) {
		switch ($mode) {
			case 'r':
			case 'rb':
				// this works if the library returns a PHP stream directly
				return $this->connection->getStream($path);
			case 'w':
			case 'w+':
			case 'wb':
			case 'wb+':
			case 'a':
			case 'ab':
			case 'r+':
			case 'a+':
			case 'x':
			case 'x+':
			case 'c':
			case 'c+':
				// most storages do not support on the fly stream upload for all modes,
				// so we use a temporary file first
				$ext = pathinfo($filename, PATHINFO_EXTENSION);
				$tmpFile = \OC::$server->getTempManager()->getTemporaryFile($ext);

				// this wrapper will call the callback whenever fclose() was called on the file,
				// after which we send the file to the library
				$result = CallbackWrapper::wrap(
					$source,
					null,
					null,
					function () use ($tmpFile, $path) {
						$this->connection->putFile($tmpFile, $path);
						unlink($tmpFile);
					}
				);
		}
		return false;
	}
```
Undefined variable $filename
Unused local variable $result

```
	public function touch($path, $time = null) {
		if ($time === null) {
			$time = time();
		}

		// many libraries might not support touch, so need to adapt
		if (!$this->file_exists($path)) {
			// create empty file
			$this->file_put_contents($path, '');
		}
		// set mtime to existing file
		return $this->connection->setModifiedTime($path, $time);
	}

	public function file_put_contents($path) {
		return $this->connection->setContents($path);
	}
```
Method call uses 2 parameters, but method signature uses 1 parameter - file_put_contents

Either:
a) fix the code to be a better example
or
b) delete the example if it is no longer relevant to anything

- [ ] also check other code examples to see if they are valid

## Where Does This Need To Be Documented?
`MyStorage.php` which will be in `modules/developer_manual/examples/app/storage-backend/OCA/MyStorageApp/Storage/MyStorage.php` after PR #756 

## What Type Of Content Change Is This? (Optional)
- [ ] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [x] Bug Fix to Existing Content

## Which Manual Does This Relate To? (Optional)
- [ ] Admin Manual
- [x] Developer Manual
- [ ] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

**Comment by mmattel** [comment]
That will be a lot of fun, see issue: https://github.com/owncloud/docs/issues/757 (Missing examples in developer docs)
There are many code snipplets in different languages to be checked and added...

**Comment by phil-davis** [comment]
After #792 is sorted out we will know what is the set of code examples that are usefully linked inot documentation. Then we can check that the examples actually have valid and up-to-date code.

**Comment by settermjd** [comment]
Thanks for creating this issue, @phil-davis. I'm having a look at the file to see what I can do to correct it. The file was originally added by Daniel Molkentin on Oct 27, 2012, and then moved around after that. Perhaps it's no longer required, given that it's that old. Do you see it as a worthwhile example, if fixed?

**Comment by phil-davis** [comment]
Actually I do not have any idea about the code samples in the docs repo. I raised the matching issue like this in the previous documentation repo because when I had it open in PHPstormIDE, the IDE found the invalid PHP for me. And it seems "not a good thing" to have invalid code in docs.

"Somebody" would have to review each code example and decide:
1) Is it useful? - if no, then delete it (deleted code is debugged code ;)
2) How to fix it? - both for obvious syntax, and also is it an up-to-date example that would work in current core+app.

Currently I am not that somebody!
